src/org/hammerhead226/sharkmacro/Parser.java

```diff
@@ -224,4 +214,24 @@ protected static int findLatestNumberedFile(String directory, String prefix) {
 		return nums;
 	}
 
+	/**
+	 * Checks if read/write directory exists on disk and creates it if it doesn't.
+	 * 
+	 * @return {@code true} if the directory already exists or was successfully
+	 *         created, {@code false} if an exception was thrown
+	 */
+	private static boolean checkDirectory(String directory) {
+		Path p = Paths.get(directory);
+		if (Files.notExists(p)) {
+			try {
+				Files.createDirectories(p);
+				System.out.println("Created Directory: " + directory);
+			} catch (IOException e) {
+				e.printStackTrace();
+				return false;
+			}
+		}
+		return true;
+	}
```

Comment:
I think `check` could be named better, maybe something like `enforce` or `touch` ([Unix](https://en.wikipedia.org/wiki/Touch_(Unix))).